docs/zkapps/o1js/smart-contracts.md

```diff
@@ -81,61 +65,41 @@ class HelloWorld extends SmartContract {
+### @method 
 
-:::
+Inside a `@method`, things sometimes behave a little differently. 
 
-<!-- TODO: create at least a basic section about zk circuits below this subsection which contains (among other things) a more thourough explanation of what this tip hints at -->
+To construct a circuit which can then be proven, o1js calls into SnarkyML, a language that builds circuits and connects variables and constraints. As a zkApp developer, you must use the methods, functions, and types provided by o1js. Plain JavaScript code does not call into SnarkyML and therefore is not able to construct circuits.
 
-:::tip
+When `SmartContract` is compiled into prover and verification keys, methods are in an environment where the method inputs don't have any concrete values attached to them. Instead, they are like mathematical variables `x`, `y`, `z` that are used to build up abstract computations like `x^2 + y^2` by running the method code.
 
-You will find that inside a `@method`, things sometimes behave a little differently. For example, the following code can't be used in a method where `x: Field` is an input parameter:
-
-```ts
-console.log(x.toString()); // don't do this inside a `@method`! 😬
-```
-
-This doesn't work because, when we compile the SmartContract into prover and verification keys, we will run your method in an environment where the method inputs don't have any concrete values attached to them. They are like mathematical variables `x`, `y`, `z` which are used to build up abstract computations like `x^2 + y^2`, just by running your method code.
-
-Therefore, when executing your code and trying to read the value of `x` to turn it into a string via `x.toString()`, it will blow up because such a value can't be found. On the other hand, during proof generation all the variables _have_ actual values attached to them (cryptographers call them "witnesses"); and it makes perfect sense to want to log these values for debugging.
-This is why we have a special function for logging stuff from inside your method:
+In contrast, all the variables _have_ actual values attached to them (cryptographers call them "witnesses") during proof generation. 
+To log these values for debugging, use a special function for logging from inside your method:
 
 ```ts
 Provable.log(x);
 ```
 
-The API is like that of `console.log`, but it will automatically handle printing o1js data types in a nice format. During SmartContract compilation, it will simply do nothing.
-:::
+The API is like `console.log`, but it automatically handles printing o1js data types in a readable format. However, it does nothing during `SmartContract` compilation.
```

Comment:
Suggestion: However, the `Provable.log(x)` function does not have any effect while the `Smart Contract` is being compiled.